File: server/configs/imageKit.js
import ImageKit from "imagekit";

let imagekit = null;

if (process.env.IMAGEKIT_PUBLIC_KEY && process.env.IMAGEKIT_PUBLIC_KEY.trim() !== "") {
    imagekit = new ImageKit({
        publicKey : process.env.IMAGEKIT_PUBLIC_KEY,
        privateKey : process.env.IMAGEKIT_PRIVATE_KEY,
        urlEndpoint : process.env.IMAGEKIT_URL_ENDPOINT
    });
} else {
    // Mock dummy imagekit to prevent startup exception
    imagekit = {
        upload: async () => {
            throw new Error("ImageKit is not configured.");
        },
        url: () => {
            return "";
        }
    };
}

export default imagekit;

File: server_python/configs/imagekit.py
# configs/imagekit.py

import os
from dotenv import load_dotenv

load_dotenv()

private_key = os.getenv("IMAGEKIT_PRIVATE_KEY")
public_key = os.getenv("IMAGEKIT_PUBLIC_KEY")
url_endpoint = os.getenv("IMAGEKIT_URL_ENDPOINT")

imagekit = None

if private_key and private_key.strip():
    try:
        from imagekitio import ImageKit
        imagekit = ImageKit(
            private_key=private_key,
            public_key=public_key,
            url_endpoint=url_endpoint
        )
    except Exception as e:
        print(f"Error initializing ImageKit: {e}")